### DISCUSSION TITLE
"Incorrect API key" error when running Structurizr Lite multiple pods

### DISCUSSION THREAD

**Discussion by claudiovianatech** [structurizr/lite#89] [Q&A] [answered]
### Question

Recurrently when I access the Structurizr Lite interface I receive the error Incorrect API key, the version data follows

It's an intermittent error, sometimes to solve it I disable the browser cache and run refresh, once or twice until the error stops happening and the workspace is loaded

Is there a config that needs to be added? Shouldn't the api key only be necessary when I connect the instance to a cloud version?

My installation uses Structurizr Lite with docker in the version below

Structurizr Lite
- build: 3135 (2023-09-05T14:18:04Z
- structurizr-java: v1.26.1
- structurizr-dsl: v1.32.0
- structurizr-import: v1.5.0
- structurizr-graphviz: v2.2.1

The error line log
2023-09-19T04:22:35.848Z ERROR 1 --- [nio-8080-exec-6] com.structurizr.lite.web.ApiController : com.structurizr.lite.web.HttpUnauthorizedException: Incorrect API key

Thanks for the help

**Comment by simonbrowndotje** [comment]
What's your workflow? Are you running a single version of Lite, multiple versions, stopping/starting Lite, etc?

**Reply by claudiovianatech** [reply]
I don't know if I understand your question correctly, but Struturize Lite is deployed in a small cluster with 2 pods.

My model development workflow is as described below:

- I make the updates locally on my machine, validate and make visual adjustments to the models running a container on localhost and after ok I update a remote repository where a github action triggers a deploy process that rebuilds the pods with the updated code.

**Reply by simonbrowndotje** [reply, marked as answer]
> Struturize Lite is deployed in a small cluster with 2 pods

Structurizr Lite isn't designed for this use case I'm afraid - each instance will generate a random API key/secret pair on startup so that the UI can communicate with the API, so this will be your issue. Your options are:

 - Modify Structurizr Lite to support HA deployments.
 - Reduce the pod size to one.
 - Use the [on-premises installation](https://docs.structurizr.com/onpremises) instead.

**Reply by claudiovianatech** [reply]
I reduced it to one pod and apparently it stopped happening... it was set to 2 as it was a default, but I don't think it's really necessary, thank you very much!
